clang/lib/CodeGen/CGAtomic.cpp

```diff
@@ -1439,6 +1439,10 @@ Address AtomicInfo::emitCastToAtomicIntPointer(Address addr) const {
     // a capability type. We also pass capabilities to the atomic libcalls
     // (i.e. always use the optimized ones) since there is no way for the
     // generic libcall to maintain tags.
+    if (!UseLibcall) {
+      // If we aren't using a libcall there is no need to cast to i8*
+      return CGF.Builder.CreateBitCast(addr, getAtomicPointer()->getType());
+    }
```

Comment:
Would be nicer to push the getPointerTo up into the cases that use it and re-use the standard return path for this case with ty (well, Ty/ValTy) set to `getAtomicPointer()->getType()`.

Reply:
Done.